### PR TITLE
Issue 437 [bugfix]: delete build artefacts in tmp folder

### PR DESCRIPTION
**Pull request by mkeuschn**
delete build artefacts in tmp folder which are causing problems in run mode

### CODE REVIEW

File: 7.0/build/s2i/bin/assemble [hunk 1]
```diff
@@ -268,3 +268,6 @@ fi
 
 # fix permissions
 fix-permissions /opt/app-root
+
+# delete build artefacts in tmp folder which are causing problems in run mode
```
Comment (tmds) [inline 7.0/build/s2i/bin/assemble:272]: You can remove the entire `/tmp/.dotnet` directory.

Can you place this a bit higher up? Just below the section that cleans up the NuGet artifacts.

**Comment by tmds** [comment]
It would be nice if we had a test for this. Could you add one?
If it's not convenient, I can provide it.

If you want to take a shot.

It needs to go in the `test/run` file.

It could be something like:
```
  # rm-tmp-dotnet creates a directory at /tmp/.dotnet/shm during assemble.
  local app=rm-tmp-dotnet

  # build image
  local image=$(s2i_image_tag ${app})
  local s2i_build=$(s2i_build_output_log ${app} ${image})
  local tmp_dotnet=$(docker run --rm ${image} stat /tmp/.dotnet 2>&1)

  # cleanup
  docker_rmi ${image}

  # verify /tmp/.dotnet was removed
  assert_contains "${tmp_dotnet}" "No such file or directory"
```

`rm-tmp-dotnet` is a folder with a `.s2i/bin/assemble` script that does something like:
```
#!/bin/bash

set -e

# generate a project
pushd /tmp/src
dotnet new web --no-restore
popd

# ensure something exists at /tmp/.dotnet
mkdir -p /tmp/.dotnet/shm

# call base assemble script
$STI_SCRIPTS_PATH/assemble
```

**Comment by tmds** [comment]
@mkeuschn is it feasible for you to add a test, otherwise I'll look into it tomorrow (before I start my Christmas holiday).

**Comment by mkeuschn** [comment]
@tmds I have no permissions to pull the rhel image. So I am not able to run the test.
```
Trying to pull docker.io/ubi8/dotnet-70:latest...
Trying to pull quay.io/ubi8/dotnet-70:latest...
Error: 2 errors occurred while pulling:
 * initializing source docker://ubi8/dotnet-70:latest: reading manifest latest in docker.io/ubi8/dotnet-70: errors:
denied: requested access to the resource is denied
unauthorized: authentication required
```

**Comment by tmds** [comment]
> @tmds I have no permissions to pull the rhel image. So I am not able to run the test.

The image needs to be built locally.

From the top-level, you can do something like:

```
IMAGE_OS=RHEL8 ./build.sh
```

That will build the images, and run the tests.

I'll give it a run on your branch.

File: 6.0/build/s2i/bin/assemble [hunk 2]
```diff
@@ -260,6 +260,9 @@ if [ "$DOTNET_INCREMENTAL" == "true" ]; then
 fi
 rm -rf /opt/app-root/{.nuget,.local}
 
+# delete build artefacts in tmp folder which are causing problems in run mode
```
Comment (tmds) [inline 6.0/build/s2i/bin/assemble:263]: nit. there is a spelling error 'artefacts' -> 'artifacts'.

**Comment by tmds** [comment]
> I'll give it a run on your branch.

This works as intended.

Thanks for implementing the fix!

@aslicerh I'll let you merge this once you had a look as well.

**Review by tmds — APPROVED** [review]

**Review by aslicerh — APPROVED** [review]
LGTM. I'll go ahead and merge it.

Thanks for the good work!

**Pull request merged**